test/e2e/suites/advanced/main_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2025 The Kube Resource Orchestrator Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package advanced
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/kro-run/kro/test/e2e/framework"
+)
+
+var (
+	f *framework.Framework
+)
+
+func TestMain(m *testing.M) {
```

Comment:
My 2c: it's much easier to understand tests that use a Harness object in the test.  The downside is that it can be slower if we're creating an envtest or kind cluster every time.